subsys/logging/log_core.c

```diff
@@ -204,7 +204,9 @@ static inline void msg_finalize(struct log_msg *msg,
 	irq_unlock(key);
 
 	if (panic_mode) {
+		key = irq_lock();
```

Comment:
Will it work if backend is interrupt driven? It can happen that locked irqs will deadlock the backend.

Reply:
when panic in log is entered then all backends are informed and they are obliged to swtich to irqless mode or shut down.